registrations/models.py

```diff
@@ -1533,36 +1537,39 @@ def partially_refund_signup_group_web_store_payment(
     def delete(self, *args, **kwargs):
         contact_person = self.actual_contact_person
         is_event_cancellation = kwargs.pop("is_event_cancellation", False)
-        payment_refunded = False
-        payment_partially_refunded = False
-        payment_cancelled = False
 
-        if settings.WEB_STORE_INTEGRATION_ENABLED:
+        payment_refunded = kwargs.pop("payment_refunded", False)
+        payment_partially_refunded = kwargs.pop("payment_partially_refunded", False)
+        payment_cancelled = kwargs.pop("payment_cancelled", False)
+        bypass_web_store_api_calls = kwargs.pop("bypass_web_store_api_calls", False)
+
+        if settings.WEB_STORE_INTEGRATION_ENABLED and not bypass_web_store_api_calls:
             payment = getattr(self, "payment", None)
-            bypass_web_store_api_calls = kwargs.pop("bypass_web_store_api_calls", False)
 
             try:
                 if not payment and (
                     signup_group := getattr(self, "signup_group", None)
                 ):
                     payment_partially_refunded = (
                         self.partially_refund_signup_group_web_store_payment(
-                            signup_group,
-                            bypass_web_store_api_calls=bypass_web_store_api_calls,
+                            signup_group
                         )
                     )
                 else:
                     payment_refunded, payment_cancelled = (
-                        self.refund_or_cancel_web_store_payment(
-                            payment,
-                            bypass_web_store_api_calls=bypass_web_store_api_calls,
-                        )
+                        self.refund_or_cancel_web_store_payment(payment)
```

Comment:
In the `refund_or_cancel_web_store_payment` there's a potential `KeyError` if the response data is wonky somehow.